### PR TITLE
auth unit test

### LINKED ISSUE #191: Authentication service test coverage
Improve test coverage to jwt and authentication service.

As we added new classes/functionality, we want to ensure software quality + unit test this work!

Leverage typescript testing (not curl). This story is closely coupled with finishing this story: https://github.com/orgs/microsoft/projects/542?pane=issue&itemId=20313667

## Acceptance Criteria
1. Code checked into samples repo.
2. Increased code coverage on authentication work & services.

### PR DESCRIPTION
**Pull request by kusweta**
Fix #191 
Added unit-test for auth folder.

### CODE REVIEW

**Review by julioalex-rezende — COMMENTED** [review]
I see tests with failure paths only in one of the files. Shouldn't we cover some failure scenarios in both options

File: data-reconciliation-app/test/unit-test/auth/validator/jwt/jwt-validation.test.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+describe("Jwt-Validator", () => {
+  let iss = "https://login.microsoftonline.com/common/v2.0";
+  let aud = MS_AAD_CONFIG.ApiIdentifierUri;
+  let appid = MS_AAD_CONFIG.ClientApplicationId;
+  let ver = "1.0";
+
+  let header = { key1: "value1" };
+  let param = { param1: "value1" };
+  let test_body: ccfapp.Body = null;
```
Comment (Aymalla) [inline data-reconciliation-app/test/unit-test/auth/validator/jwt/jwt-validation.test.ts:16]: If these variables will not be changed in later stage, we could define it as `const`

**Review by Aymalla — APPROVED** [review]
LGTM, just nit comment added

File: data-reconciliation-app/test/unit-test/auth/validator/jwt/demo-jwt-provider.test.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+describe("Demo-Jwt-Provider", () => {
+  let sub: string = "test";
+  let iss = "https://demo";
+
+  const jwt = { header: "test", keyIssuer: iss, payload: { sub: sub } };
+  const identity: ccfapp.JwtAuthnIdentity = { policy: "jwt", jwt: jwt };
+
+  beforeEach(() => {});
+  afterEach(() => {});
```
[the remark this answers is not in the document]
Comment (kusweta) [reply data-reconciliation-app/test/unit-test/auth/validator/jwt/demo-jwt-provider.test.ts]: have removed it from individual files.

**Pull request merged**